indico/modules/rb/models/reservation_occurrences.py

```diff
@@ -289,22 +289,40 @@ def is_in_notification_window(self, exclude_first_day=False):
         if self.start_dt.date() < date.today():
             return False
         days_until_occurrence = (self.start_dt.date() - date.today()).days
-        notification_window = (self.reservation.room.notification_before_days
-                               or rb_settings.get('notification_before_days', 1))
+        if self.reservation.is_repeating:
+            notification_window = (self.reservation.room.notification_before_repeating or
+                                   rb_settings.get('notification_before_repeating'))
+        else:
+            notification_window = (self.reservation.room.notification_before_single or
+                                   rb_settings.get('notification_before_single'))
         if exclude_first_day:
             return days_until_occurrence < notification_window
         else:
             return days_until_occurrence <= notification_window
 
     @is_in_notification_window.expression
-    def is_in_notification_window(self, exclude_first_day=False):
+    def is_in_notification_window(cls, exclude_first_day=False):
+        from indico.modules.rb import Reservation
         from indico.modules.rb import settings as rb_settings
+        from indico.modules.rb.models.reservations import RepeatFrequency
         from indico.modules.rb.models.rooms import Room
-        in_the_past = cast(self.start_dt, Date) < cast(func.now(), Date)
-        days_until_occurrence = cast(self.start_dt, Date) - cast(func.now(), Date)
-        notification_window = func.coalesce(Room.notification_before_days,
-                                            rb_settings.get('notification_before_days', 1))
+        in_the_past = cast(cls.start_dt, Date) < cast(func.now(), Date)
+        days_until_occurrence = cast(cls.start_dt, Date) - cast(func.now(), Date)
```

Comment:
Why not storing the output of `cast(cls.start_dt, Date)` and `cast(func.now(), Date)` into variables? You use `cast` 4 times instead of 2 :smile: